### ISSUE TITLE
PhotonImage.new_from_blob does not work

### ISSUE THREAD

**Issue by sabercoy** [fineshopdesign/cf-wasm#40]
running this in wrangler gives

[wrangler:err] RuntimeError: unreachable

**Comment by fineshop** [comment]
I am not familiar with rust programming language.
It looks like [silvia-odwyer/photon](https://github.com/silvia-odwyer/photon)'s `PhotonImage.new_from_blob` function is trying to create an `Uint8Array` instance directly from `Blob`, i.e. `new Uint8Array(blob)`

https://github.com/silvia-odwyer/photon/blob/941adf9d2a60f9b92a2b8016b5036033dbffb192/crate/src/lib.rs#L107C1-L115C6

Can you create an issue there if I am not wrong?

**Comment by fineshop** [comment]
Created an issue https://github.com/silvia-odwyer/photon/issues/197

**Comment by sabercoy** [comment]
ChatGPT says the Rust code is incorrect and it needs to be converted to an ArrayBuffer first
Which is funny, because that's what happens in JS land
This makes me wonder why this method exists, unless its significantly faster to make ArrayBuffers/Uint8Arrays in Rust

**Comment by fineshop** [comment]
Yes, it is same as converting blob to arraybuffer then Uint8Array, and using 
```ts
const blob = /* Blob */
const buffer = await blob.arrayBuffer();
const bytes = new Uint8Array(buffer);
const inputImg = Photon.new_from_byteslice(bytes);
```

**Comment by cyyyu** [comment]
@sabercoy did you find a solution?

**Comment by kumardeo** [comment]
You can use the following instead:

```
const blob = /* Blob */;
const buffer = await blob.arrayBuffer();
const bytes = new Uint8Array(buffer);
const inputImg = Photon.new_from_byteslice(bytes);
```

**Comment by cyyyu** [comment]
@kumardeo thx. but I've tried. it doesn't work for me, as inside `new_from_blob` it does the same thing.

**Comment by sabercoy** [comment]
> [@sabercoy](https://github.com/sabercoy) did you find a solution?

It has been a while since I attempted this. I believe I did the converting to ArrayBuffer as a workaround. In the end, the library was too slow for my use case so I have moved on to a browser-side solution.

**Comment by kumardeo** [comment]
> [@sabercoy](https://github.com/sabercoy) that's a good move. It's interesting I ended up with the same strategy like you did just now. It wasted a day of mine to find workarounds but none of them worked.

I believe there is issue in implementation of `Photon.new_from_blob` in upstream repository.
You can instead use `Photon.new_from_byteslice` as suggested by @sabercoy. `Photon.new_from_blob` internally uses `Photon.new_from_byteslice` so I guess there is no difference in the followings:

```ts
const blob = /* Blob */;
const bytes = new Uint8Array(await blob.arrayBuffer());
const inputImg = Photon.new_from_byteslice(bytes);
```
should be same as
```ts
const blob = /* Blob */;
const inputImg = Photon.new_from_blob(blob);
```
